app/controllers/downloads_controller.rb

```diff
@@ -2,7 +2,7 @@ class DownloadsController < ApplicationController
 
   skip_before_action :store_location, only: :show
   before_action :load_work, only: :show
-  before_action :check_visibility, only: :show
+  before_action :authenticate_user!, only: :show
```

Comment:
`check_visibility` handles a few more things that `authenticate_user!` doesn't:

- if the work is a draft
- if the work is hidden by admin

This check also seemingly means that you cannot download any work at all unless logged in. Logged out users should be able to get at unrestricted works.

Reply:
Okay, back to the other plan of adding another branch to check_visibility, then!